Clarify statement of need (#658)
trixi-framework/TrixiParticles.jl: README.md

[![License: MIT](https://img.shields.io/badge/License-MIT-success.svg)](https://opensource.org/licenses/MIT)
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.10797541.svg)](https://zenodo.org/doi/10.5281/zenodo.10797541)

**TrixiParticles.jl** is a high-performance numerical simulation framework for particle-based methods, focused on the simulation of complex multiphysics problems, and written in [Julia](https://julialang.org).

> TrixiParticles.jl focuses on the following use cases:
> - Development of new particle-based methods and models.
> - Accurate and efficient physics-based modelling of complex multiphysics problems.
> - Easy setup of accessible simulations for educational purposes, including student projects, coursework, and thesis work.

Reviewer: ```suggestion
TrixiParticles.jl primarily focuses on the following use cases:
- Accurate and efficient physics-based modelling of complex multiphysics problems.
- Development of new particle-based methods and models.
- Easy setup of accessible simulations for educational purposes, including student projects, coursework, and thesis work.
```
I would put multiphysics first (since it's part of what sets it apart from your standard SPH code)

It offers intuitive configuration, robust pre- and post-processing, and vendor-agnostic GPU-support based on the Julia package [KernelAbstractions.jl](https://github.com/JuliaGPU/KernelAbstractions.jl). 

> In the future we plan to enable large-scale simulations through adding MPI support.

Reviewer: ```suggestion
```
I wouldn't put this here until it's actually available.

Author: I mostly added this since the question regarding "need" came up in the JOSS review. But we can also leave it out for now.

Reviewer: Usually I am not a big fan of "announcing" planned features, but rather just have them and then report on them :-)

[![License: MIT](https://img.shields.io/badge/License-MIT-success.svg)](https://opensource.org/licenses/MIT)
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.10797541.svg)](https://zenodo.org/doi/10.5281/zenodo.10797541)

> **TrixiParticles.jl** is a high-performance particle simulation solver framework written in [Julia](https://julialang.org). It offers intuitive configuration, robust pre- and post-processing, and seamless scalability from CPU to Exascale-level GPU computing, using KernelAdaptations.jl for GPU manufacturer-agnostic support and MPI (WIP) for parallel processing. The extensible architecture allows users to incorporate additional particle methods easily, making TrixiParticles.jl ideal for research and educational purposes.

Reviewer 2: ```suggestion
**TrixiParticles.jl** is a high-performance particle simulation solver framework written in [Julia](https://julialang.org). It offers intuitive configuration, robust pre- and post-processing, and seamless scalability from CPU to Exascale-level GPU computing, using KernelAbstractions.jl for GPU manufacturer-agnostic support and MPI (WIP) for parallel processing. The extensible architecture allows users to incorporate additional particle methods easily, making TrixiParticles.jl ideal for research and educational purposes.
```

> **TrixiParticles.jl** is a high-performance particle simulation solver framework written in [Julia](https://julialang.org). It offers intuitive configuration, robust pre- and post-processing, and seamless scalability from CPU to Exascale-level GPU computing, using KernelAbstractions.jl for GPU manufacturer-agnostic support and MPI (WIP) for parallel processing. The extensible architecture allows users to incorporate additional particle methods easily, making TrixiParticles.jl ideal for research and educational purposes.

Reviewer 3: 1. "particle simulation solver framework" sounds weird IMO, and it sounds like we are simulating particles, which is misleading.
2. We don't have MPI on main and "seamless scalability from CPU to Exascale-level GPU computing" is a vast overstatement.
3. It does not answer the question in #630 about the "target audience". I tried to add this below:
```suggestion
**TrixiParticles.jl** is a high-performance numerical simulation framework using particle-based numerical methods to simulate complex multiphysics problems and it is written in [Julia](https://julialang.org).
TrixiParticles.jl is targeted towards two groups of users:
- Users researching particle-based methods who want to develop new techniques or modify existing schemes.
- Users who want to simulate complex multiphysics and do not necessarily interested in the underlying numerical methods.

It offers intuitive configuration, robust pre- and post-processing, and vendor-agnostic GPU-support based on the Julia package [KernelAbstractions.jl](https://github.com/JuliaGPU/KernelAbstractions.jl). The extensible architecture allows users to incorporate additional particle methods easily, making TrixiParticles.jl ideal for research and educational purposes.
```
